### PR TITLE
cnf-network: add mlxSecureBoot test

### PR DESCRIPTION
**Pull request by evgenLevin**
https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-77014&revision=14626915

### CODE REVIEW

File: tests/cnf/core/network/sriov/tests/externalllymanaged.go [hunk 1]
```diff
@@ -63,7 +63,8 @@ var _ = Describe("ExternallyManaged", Ordered, Label(tsparams.LabelExternallyMan
 				Expect(err).ToNot(HaveOccurred(), "Failed to retrieve SR-IOV interfaces for testing")
 
 				if sriovenv.IsMellanoxDevice(sriovInterfacesUnderTest[0], workerNodeList[0].Object.Name) {
-					configureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], true, 5)
+					err = sriovenv.ConfigureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], true, 5)
+					Expect(err).ToNot(HaveOccurred(), "Failed to cofigure Mellanox firmware")
```
Comment (ajaggapa) [inline tests/cnf/core/network/sriov/tests/externalllymanaged.go:67]: typo: configure

File: tests/cnf/core/network/sriov/tests/externalllymanaged.go [hunk 2]
```diff
@@ -110,7 +111,8 @@ var _ = Describe("ExternallyManaged", Ordered, Label(tsparams.LabelExternallyMan
 				Expect(err).ToNot(HaveOccurred(), "Failed to remove all NMState policies")
 
 				if sriovenv.IsMellanoxDevice(sriovInterfacesUnderTest[0], workerNodeList[0].Object.Name) {
-					configureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], false, 0)
+					err = sriovenv.ConfigureSriovMlnxFirmwareOnWorkersAndWaitMCP(workerNodeList, sriovInterfacesUnderTest[0], false, 0)
+					Expect(err).ToNot(HaveOccurred(), "Failed to cofigure Mellanox firmware")
```
Comment (ajaggapa) [inline tests/cnf/core/network/sriov/tests/externalllymanaged.go:115]: typo

File: tests/cnf/core/network/internal/netconfig/config.go [hunk 3]
```diff
@@ -185,6 +188,37 @@ func (netConfig *NetworkConfig) GetSwitchLagNames() ([]string, error) {
 	return envValue, nil
 }
 
+// GetBMCHostNames checks the environmental variable ECO_CNF_CORE_NET_BMC_HOST_NAMES
+// and returns the value in []string.
+func (netConfig *NetworkConfig) GetBMCHostNames() ([]string, error) {
+	envValue := strings.Split(netConfig.BMCHostNames, ",")
+
+	if len(envValue) == 0 {
```
[the remark this answers is not in the document]
Comment (evgenLevin) [reply tests/cnf/core/network/internal/netconfig/config.go]: fixed

File: tests/cnf/core/network/sriov/tests/externalllymanaged.go (hunk 1 above)
Comment (evgenLevin) [reply tests/cnf/core/network/sriov/tests/externalllymanaged.go]: fixed

File: tests/cnf/core/network/sriov/tests/externalllymanaged.go (hunk 2 above)
Comment (evgenLevin) [reply tests/cnf/core/network/sriov/tests/externalllymanaged.go]: fixed

**Review by ajaggapa — APPROVED** [review]
lgtm

**Comment by kononovn** [comment]
@evgenLevin Please update PR title/commit message to be align with https://github.com/openshift-kni/eco-gotests#commit-message-guidelines

**Comment by evgenLevin** [comment]
> @evgenLevin Please update PR title/commit message to be align with https://github.com/openshift-kni/eco-gotests#commit-message-guidelines

Changed the PR tittle

File: tests/cnf/core/network/sriov/tests/mlxSecureBoot.go [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+				sriovAndResourceNameSecureBoot,
+				2,
+				sriovInterfacesUnderTest[:1],
+				map[string]string{"kubernetes.io/hostname": workerNodeList[0].Definition.Name})
+
+			err := sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(sriovPolicy, tsparams.MCOWaitTimeout)
+			Expect(err).ToNot(HaveOccurred(), "Failed to configure SR-IOV policy")
+
+			By("Creating SR-IOV network")
+
```
Comment (SchSeba) [inline tests/cnf/core/network/sriov/tests/mlxSecureBoot.go:118]: nit: remove empty line

File: tests/cnf/core/network/sriov/tests/mlxSecureBoot.go [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+			Expect(err).ToNot(HaveOccurred(), "Failed to discover SR-IOV node state")
+
+			currentTotalVFs, err := sriovNodeState.GetTotalVFs(sriovInterfacesUnderTest[0])
+			Expect(err).ToNot(HaveOccurred(), "Failed to get totalvfs on SR-IOV node")
+
+			Expect(currentTotalVFs).To(BeNumerically("==", totalVFs))
+		})
+	})
+
+func configureSecureBoot(bmcClient *bmc.BMC, action string) {
```
Comment (SchSeba) [inline tests/cnf/core/network/sriov/tests/mlxSecureBoot.go:153]: Should these two functions be moved out of the test file?

File: tests/cnf/core/network/sriov/tests/mlxSecureBoot.go [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+			sriovPolicy := sriov.NewPolicyBuilder(
+				APIClient,
+				sriovAndResourceNameSecureBoot,
+				NetConfig.SriovOperatorNamespace,
+				sriovAndResourceNameSecureBoot,
+				2,
+				sriovInterfacesUnderTest[:1],
+				map[string]string{"kubernetes.io/hostname": workerNodeList[0].Definition.Name})
+
+			err := sriovenv.CreateSriovPolicyAndWaitUntilItsApplied(sriovPolicy, tsparams.MCOWaitTimeout)
```
Comment (SchSeba) [inline tests/cnf/core/network/sriov/tests/mlxSecureBoot.go:114]: one more verification here. can you check that the node didn't reboot?
you can get the number of restarts on the config-daemon pod on the node and check it's equal after the MCP and nodeState are stable

File: tests/cnf/core/network/sriov/tests/mlxSecureBoot.go (hunk 4 above)
Comment (evgenLevin) [reply tests/cnf/core/network/sriov/tests/mlxSecureBoot.go:133]: removed

File: tests/cnf/core/network/sriov/tests/mlxSecureBoot.go (hunk 5 above)
Comment (evgenLevin) [reply tests/cnf/core/network/sriov/tests/mlxSecureBoot.go]: moved

**Review by SchSeba — APPROVED** [review]
LGTM

**Review by ajaggapa — APPROVED** [review]
lgtm

**Review by ajaggapa — APPROVED** [review]
lgtm

**Review by gkopels — APPROVED** [review]

**Comment by gkopels** [comment]
lgtm

**Pull request merged**